src/txadmission.cpp

```diff
@@ -31,6 +32,13 @@
 
 using namespace std;
 
+static void TestConflictEnqueueTx(CTxInputData &txd);
+
+// The average commit batch size is used to limit the quantity of transactions that are moved from the defer queue
+// onto the inqueue.  Without this, if received transactions far outstrip processing capacity, transactions can be
+// shuffled between the in queue and the defer queue with little progress being made.
+const uint64_t minCommitBatchSize = 10000;
+uint64_t avgCommitBatchSize = 0;
```

Comment:
What lock is protecting `avgCommitBatchSize`?

Reply:
csCommitQ for modifications, and I've changed it to an atomic for reads.